pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -54,230 +51,114 @@ func (e transientError) Error() string {
+		for _, deployment := range existingDeployments.Items {
+			oldReplicaCount := deployment.Spec.Replicas
+			newReplicaCount := oldReplicaCount
+			// The active deployment should be scaled to the config replica count, and
+			// everything else should be at zero.
+			if activeDeployment != nil && deployment.Name == activeDeployment.Name {
+				newReplicaCount = config.Template.ControllerTemplate.Replicas
+			} else {
+				newReplicaCount = 0
 			}
-			// Try to see if the image stream exists, if not then the build will never be able to update the
-			// tag in question
-			if _, err := c.osClient.ImageStreams(config.Namespace).Get(name); err != nil {
-				glog.V(2).Infof("Error while trying to get image stream %q: %v", name, err)
-				if errors.IsNotFound(err) {
-					details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream %q does not exist.", istag, name)
+			// Only process updates if the replica count actually changed.
+			if oldReplicaCount != newReplicaCount {
+				deployment.Spec.Replicas = newReplicaCount
+				_, err := c.kubeClient.ReplicationControllers(deployment.Namespace).Update(&deployment)
+				if err != nil {
+					c.recorder.Eventf(config, "DeploymentScaleFailed", "Failed to scale deployment %q from %d to %d: %s", deployment.Name, oldReplicaCount, newReplicaCount, err)
+					return err
 				}
+				c.recorder.Eventf(config, "DeploymentScaled", "Scaled deployment %q from %d to %d", deployment.Name, oldReplicaCount, newReplicaCount)
 			}
 		}
-		allDetails = append(allDetails, details)
-	}
-
-	if len(allDetails) > 1 {
-		for i := range allDetails {
-			allDetails[i] = fmt.Sprintf("\t* %s", allDetails[i])
-		}
-		// Prepend multiple errors warning
-		multipleErrWarning := fmt.Sprintf("Deployment config %q blocked by multiple errors:\n", config.Name)
-		allDetails = append([]string{multipleErrWarning}, allDetails...)
+		return nil
 	}
-
-	return strings.Join(allDetails, "\n"), existingDeployments, latestDeploymentExists, nil
-}
-
-// updateDetails updates a deployment configuration with the provided details
-func (c *DeploymentConfigController) updateDetails(config *deployapi.DeploymentConfig, details string) (*deployapi.DeploymentConfig, error) {
-	if config.Details == nil {
-		config.Details = new(deployapi.DeploymentDetails)
+	// No deployments are running and the latest deployment doesn't exist, so
+	// create the new deployment.
+	deployment, err := deployutil.MakeDeployment(config, c.codec)
+	if err != nil {
+		return fatalError(fmt.Sprintf("couldn't make deployment from (potentially invalid) deployment config %s: %v", deployutil.LabelForDeploymentConfig(config), err))
 	}
-	if details != config.Details.Message {
-		config.Details.Message = details
-		return c.osClient.DeploymentConfigs(config.Namespace).Update(config)
+	created, err := c.kubeClient.ReplicationControllers(config.Namespace).Create(deployment)
```

Comment:
Is there any particular reason you are setting the config namespace here vs in the utility? I would expect that creating a deployment from a config would always set a namespace to the deployment (inherited by the config).